kernel/src/engine/arrow_utils.rs

```diff
@@ -306,16 +308,23 @@ fn get_indices(
     // for each field, get its position in the parquet (via enumerate), a reference to the arrow
     // field, and info about where it appears in the requested_schema, or None if the field is not
     // requested
-    let all_field_info = fields.iter().enumerate().map(|(parquet_index, field)| {
-        let field_info = requested_schema.fields.get_full(field.name());
-        (parquet_index, field, field_info)
-    });
-    for (parquet_index, field, field_info) in all_field_info {
+    let matched_parquet_fields = match_parquet_fields(requested_schema, fields);
+    for MatchedParquetField {
```

Comment:
nit: I like to define structs before use, helps with code reading (so move the def of `MatchedParquetField` up)